src/super_gradients/training/utils/sg_trainer_utils.py

```diff
@@ -190,28 +190,26 @@ def update_monitored_values_dict(monitored_values_dict: Dict[str, MonitoredValue
     :param new_values_dict: Dict mapping value names to their new (i.e. current epoch) value.
     :return: Updated monitored_values_dict
     """
-    for monitored_value_name in monitored_values_dict.keys():
+    relevant_keys = set(new_values_dict.keys()).intersection(monitored_values_dict.keys())
+    for monitored_value_name in relevant_keys:
+        previous_value = monitored_values_dict[monitored_value_name]
         monitored_values_dict[monitored_value_name] = update_monitored_value(
             new_value=new_values_dict[monitored_value_name],
-            previous_monitored_value=monitored_values_dict[monitored_value_name],
+            previous_monitored_value=previous_value,
         )
     return monitored_values_dict
 
 
-def display_epoch_summary(
-    epoch: int, n_digits: int, train_monitored_values: Dict[str, MonitoredValue], valid_monitored_values: Dict[str, MonitoredValue]
-) -> None:
+def display_epoch_summary(epoch: int, n_digits: int, monitored_values_dict: Dict[str, Dict[str, MonitoredValue]]) -> None:
     """Display a summary of loss/metric of interest, for a given epoch.
 
     :param epoch: the number of epoch.
     :param n_digits: number of digits to display on screen for float values
-    :param train_monitored_values: mapping of loss/metric with their stats that will be displayed
-    :param valid_monitored_values: mapping of loss/metric with their stats that will be displayed
-    :return:
+    :param monitored_values_dict: Dict of Dict. The first one represents the splut, and the second one a loss/metric.
     """
 
     def _format_to_str(val: float) -> str:
```

Comment:
Maybe change `val: float` to `val: Optional[float]` since we now allow val to be None